packages/shared-types/src/index.ts

```diff
@@ -0,0 +1,164 @@
+// User-related types
+export interface User {
+  id: string;
+  createdAt: Date;
+  updatedAt: Date;
+  userAccounts: UserAccount[];
+  devices: Device[];
+}
+
+export interface UserAccount {
+  id: string;
+  accountId: string;
+  createdAt: Date;
+  userId: string;
+}
+
+export interface Device {
+  id: string;
+  deviceToken: string;
+  deviceType: DeviceType;
+  isActive: boolean;
+  createdAt: Date;
+  updatedAt: Date;
+  userId: string;
+}
+
+export type DeviceType = 'MOBILE' | 'WEB';
+
+// Swap-related types
+export interface Swap {
+  id: string;
+  swapId: string;
+  sellAsset: string;
+  buyAsset: string;
+  sellAmountCryptoBaseUnit: string;
+  expectedBuyAmountCryptoBaseUnit: string;
+  sellAmountCryptoPrecision: string;
+  expectedBuyAmountCryptoPrecision: string;
+  actualBuyAmountCryptoPrecision?: string;
+  status: SwapStatus;
+  source: string;
+  swapperName: string;
+  sellAccountId: string;
+  buyAccountId?: string;
+  receiveAddress?: string;
+  sellTxHash?: string;
+  buyTxHash?: string;
+  txLink?: string;
+  statusMessage?: string;
+  isStreaming: boolean;
+  estimatedCompletion?: Date;
+  createdAt: Date;
+  updatedAt: Date;
+  metadata: string;
+  chainflipSwapId?: number;
+  relayTransactionMetadata?: string;
+  relayerExplorerTxLink?: string;
+  relayerTxHash?: string;
+  stepIndex: number;
+  streamingSwapMetadata?: string;
+  userId: string;
+  notifications: Notification[];
+}
+
+export type SwapStatus = 'IDLE' | 'PENDING' | 'SUCCESS' | 'FAILED';
+
+// Notification types
+export interface Notification {
+  id: string;
+  title: string;
+  body: string;
+  type: NotificationType;
+  isRead: boolean;
+  sentAt: Date;
+  deliveredAt?: Date;
```

Comment:
q: do we need this distinction ? (just starting to read this PR out, we may or may not)

Reply:
We probably don't, at least for now, removed